tests/phpunit/elementor/core/upgrade/test-upgrades.php

```diff
@@ -453,4 +474,52 @@ private function create_image() {
 	private function delete_image( $attachment_id ) {
 		wp_delete_attachment( $attachment_id, true );
 	}
-}
+
+	/**
+	 * @param $documents
+	 *
+	 * @return void
+	 */
+	public function assert_sections_not_changed( $elementor_data ) {
+		$top_level_element = $elementor_data[0];
+		self::assertFalse( $top_level_element->isInner );
+		self::assertFalse( $top_level_element->elements[0]->isInner );
+	}
+
+	/**
+	 * @param $documents
+	 *
+	 * @return void
+	 */
+	public function assert_container_changed( $elementor_data ) {
+		$top_level_container = $elementor_data[0];
+		self::assertFalse( $top_level_container->isInner );
+		self::assertTrue( $top_level_container->elements[0]->isInner );
+		self::assertTrue( $top_level_container->elements[0]->elements[0]->isInner );
+		self::assertTrue( $top_level_container->elements[0]->elements[1]->isInner );
+	}
+
+	/**
+	 * @param $documents
+	 *
+	 * @return void
+	 */
+	private function assert_nested_elements_not_affected( $elementor_data ) {
+		$top_level_container = $elementor_data[0];
+		self::assertFalse( $top_level_container->isInner );
+		$widget = $top_level_container->elements[0];
+		self::assertFalse( $widget->elements[0]->isInner );
+		self::assertFalse( $widget->elements[1]->isInner );
+		self::assertFalse( $widget->elements[2]->isInner );
+	}
+
+	/**
+	 * @param $documents
+	 *
+	 * @return mixed
+	 */
+	public function get_decoded_post_meta( $documents ) {
+		$post_meta = get_post_meta( $documents->get_main_id(), '_elementor_data' );
+		return json_decode( $post_meta[0] );
+	}
```

Comment:
If you can use` $document->get_json_meta( '_elementor_data' );` then this function can be removed.